### PR TITLE
MODSOURMAN-1022: remove step of initial saving of incoming records to SRS

### PR DESCRIPTION
**Pull request by yaroslav-epam**
## Purpose
Remove step of initial saving of incoming records to SRS

## Approach
Removing step of initial saving of incoming records to SRS and using `RecordsPublishingServiceImpl` to send events into `DI_INCOMING_EDIFACT_RECORD_PARSED` and `DI_INCOMING_MARC_BIB_RECORD_PARSED` topics

[MODSOURMAN-1022](https://issues.folio.org/browse/MODSOURMAN-1022)

### CODE REVIEW

**Review by Maksat-Galymzhan — APPROVED** [review]

File: mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java [hunk 1]
```diff
@@ -220,6 +209,38 @@ public Future<List<Record>> parseRawRecordsChunkForJobExecution(RawRecordsDto ch
...
+    saveRecords(params, jobExecution, parsedRecords)
+      .onComplete(postAr -> {
+        if (postAr.failed()) {
+          StatusDto statusDto = new StatusDto()
+            .withStatus(StatusDto.Status.ERROR)
+            .withErrorStatus(StatusDto.ErrorStatus.RECORD_UPDATE_ERROR);
+          jobExecutionService.updateJobExecutionStatus(jobExecution.getId(), statusDto, params)
+            .onComplete(r -> {
+              if (r.failed()) {
+                LOGGER.warn("parseRawRecordsChunkForJobExecution:: Error during update jobExecution and snapshot status", r.cause());
```
Comment (VRohach) [inline mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java:222]: @yaroslav-epam Let's log here jobExecutionId also.

File: mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java [hunk 2]
```diff
@@ -220,6 +209,38 @@ public Future<List<Record>> parseRawRecordsChunkForJobExecution(RawRecordsDto ch
...
+              }
+            });
+          jobExecutionSourceChunkDao.getById(sourceChunkId, params.getTenantId())
+            .compose(optional -> optional
+              .map(sourceChunk -> jobExecutionSourceChunkDao
+                .update(sourceChunk.withState(JobExecutionSourceChunk.State.ERROR), params.getTenantId()))
+              .orElseThrow(() -> new NotFoundException(String.format(
+                "Couldn't update failed jobExecutionSourceChunk status to ERROR, jobExecutionSourceChunk with id %s was not found",
```
Comment (VRohach) [inline mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java:230]: @yaroslav-epam  The same.

File: mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java [hunk 3]
```diff
@@ -220,6 +209,38 @@ public Future<List<Record>> parseRawRecordsChunkForJobExecution(RawRecordsDto ch
...
+                sourceChunkId))))
+            .onComplete(ar -> promise.fail(postAr.cause()));
+        } else {
+          promise.complete(parsedRecords);
+        }
+      });
+  }
+
+  private Future<Boolean> sendEvents(List<Record> records, JobExecution jobExecution, OkapiConnectionParams params, DataImportEventTypes eventType) {
+    LOGGER.info("sendEvents:: Sending events with type: {}", eventType.value());
```
Comment (VRohach) [inline mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java:240]: @yaroslav-epam  The same.

**Review by VRohach — APPROVED** [review]

File: mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java [hunk 4]
```diff
@@ -181,37 +184,23 @@ public Future<List<Record>> parseRawRecordsChunkForJobExecution(RawRecordsDto ch
         if (updateMarcActionExists(jobExecution) || updateInstanceActionExists(jobExecution) || isCreateOrUpdateItemOrHoldingsActionExists(jobExecution, parsedRecords)) {
```
Comment (Aliaksandr-Fedasiuk) [inline mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java:184]: not enough log messages

File: mod-source-record-manager-server/src/test/java/org/folio/verticle/consumers/RawMarcChunkConsumersVerticleTest.java [hunk 5]
```diff
@@ -279,7 +275,7 @@ public void shouldNotObserveValuesWhenJobExecutionIdNotCreated() throws Interrup
   }
 
   @Test
-  public void shouldNotSendAnyEventsForDuplicates() throws InterruptedException {
+  public void shouldSendDIErrorForDuplicatesWhenParsedRecordIsNull() throws InterruptedException {
```
Comment (RuslanLavrov) [inline mod-source-record-manager-server/src/test/java/org/folio/verticle/consumers/RawMarcChunkConsumersVerticleTest.java:278]: seems we need to get this test back because event should not be sent when a duplicate chunk is received

**Review by Aliaksandr-Fedasiuk — APPROVED** [review]

**Review by RuslanLavrov — APPROVED** [review]
